core/providers/openai.go

```diff
@@ -221,11 +221,18 @@ func prepareOpenAIChatRequest(messages []schemas.BifrostMessage, params *schemas
 		} else if (msg.UserMessage != nil && msg.UserMessage.ImageContent != nil) || (msg.ToolMessage != nil && msg.ToolMessage.ImageContent != nil) {
 			var messageImageContent schemas.ImageContent
 			if msg.UserMessage != nil && msg.UserMessage.ImageContent != nil {
+				// Create a copy to avoid modifying the original
 				messageImageContent = *msg.UserMessage.ImageContent
 			} else if msg.ToolMessage != nil && msg.ToolMessage.ImageContent != nil {
+				// Create a copy to avoid modifying the original
 				messageImageContent = *msg.ToolMessage.ImageContent
 			}
 
+			// Normalize content first (type inference and media type normalization)
```

Comment:
@Pratham-Mishra04 the better signature here is 

```
messageImageContent.Normalize()
```

Reply:
okay